deps/db/src/logseq/db/common/initial_data.cljs

```diff
@@ -132,27 +128,35 @@
 
 (defn get-block-children-ids
   "Returns children UUIDs"
-  [db block-uuid]
+  [db block-uuid & {:keys [include-collapsed-children?]
+                    :or {include-collapsed-children? true}}]
   (when-let [eid (:db/id (d/entity db [:block/uuid block-uuid]))]
-    (let [seen   (volatile! [])]
+    (let [seen (volatile! #{})]
       (loop [steps          100      ;check result every 100 steps
              eids-to-expand [eid]]
         (when (seq eids-to-expand)
-          (let [eids-to-expand*
-                (mapcat (fn [eid] (map first (d/datoms db :avet :block/parent eid))) eids-to-expand)
-                uuids-to-add (remove nil? (map #(:block/uuid (d/entity db %)) eids-to-expand*))]
+          (let [children
+                (mapcat (fn [eid]
+                          (let [e (d/entity db eid)]
+                            (when (or include-collapsed-children?
+                                      (not (:block/collapsed? e))
+                                      (common-entity-util/page? e))
+
+                              (:block/_parent e)))) eids-to-expand)
+                uuids-to-add (keep :block/uuid children)]
             (when (and (zero? steps)
                        (seq (set/intersection (set @seen) (set uuids-to-add))))
```

Comment:
I don't understand why we check a small amount of data every 100 steps here

Reply:
Yes, it doesn't work for less than 100 steps, but It does test all the data because all the visited children have been added to `seen`. How about removing the check here?

Reply:
> but It does test all the data because all the visited children have been added to seen

but it only tested `uuids-to-add` of a certain loop...

I agree to remove it

Reply:
[Removed](https://github.com/logseq/logseq/commit/8dfec94266f191eff1e01ff3ce55ae8940f360ca).